### ISSUE TITLE
Form file field doesn't work with another disk.

### ISSUE THREAD

**Issue by OppPMatos** [z-song/laravel-admin#5213]
```php
$form->image('image')->disk('oss');
$form->file('file')->disk('local');
```

用disk方法指定要保存的disk

_Originally posted by @z-song in https://github.com/z-song/laravel-admin/issues/5090#issuecomment-723829477_

Hello.
This doesn't seem to be working for me.

I've got:
Laravel Version: 6.18.43
PHP Version: 7.x
Laravel-admin: 1.8.1

I'm using the file field and after adding the disk, nothing changes.
I've inspected the html and the url is still relative to the current site and not the the disk location.
I've also tested the disk configuration on Tinker and it's correct.

Can someone please point me in the right direction?

Edit:
I'm trying to use and FTP connection for the file.
Tried force the root directory of the ftp account to avoid navigation or any type of configuration of the field.

**Comment by AnupDPokhrel** [comment]
check if you can put your file in another disk using storage facade. If everything works OK then further investigation needs to be done.
Check and start doing dd in these files, ...

```
vendor/encore/laravel-admin/src/Form/Field/UploadField.php 
vendor/encore/laravel-admin/src/Form/Field/File.php
```

**Comment by OppPMatos** [comment]
With further testing, I found that the file field is not fully compatible with an FTP disk, and a couple of issues.
I had to use the move() property to properly place the file on the expected place, since if nothing is done the file will be uploaded to a new folder called 'files' on the configured 'root' of the ftp disk 
That's a problem, because if you configure the disk's root to be the end destination, not using the move() property, will cause the file to be placed on a 'files' folder, inheriting the laravel-admin configuration for the file upload folder name, this makes no sense, since the inheritance shouldn't exists if we on a custom disk.
But even then, after uploading the file, when trying to edit the record the field doesn't work as expected.
Because the file field will try to preview a file on a relative location to the site itself, not to the used disk.
This makes managing remote sites pretty much unbearable.
To solve this, the config/admin.php would need extra configurations to merge with the FTP disk configuration, where we could enter the alternate prefix (http) when editing the the record, making the file preview work.
